File: app/be/src/models/app_state.py
from sqlalchemy import JSON, Column, Integer, Enum

from common_python.dto import KafkaMq135, KafkaDht22
from common_python.model import PydanticJson
# from dto.alarm import Alarm
from util.enums import AppMode
from util.database import BaseSqlModel


class AppState(BaseSqlModel):
    __tablename__ = "state"

    id = Column(Integer, primary_key=True, index=False)
    current_mode = Column(Enum(AppMode))
    servo_multiple = Column(Integer)
    active_alarms = Column(JSON)
    """
    List of dto.Alarms
    """
    dht22_statistics = Column(PydanticJson(KafkaDht22))
    mq135_statistics = Column(PydanticJson(KafkaMq135))

File: app/fe/src/types/AppState.ts
import type Alarm from "./Alarms";
import type AppMode from "./AppMode";
import type Dht22Kafka from "./Dht22Kafka";
import type Mq135Kafka from "./Mq135Kafka";

interface AppState {
	current_mode: AppMode;
	servo_multiple: number;
	active_alarms: Alarm[];
	dht22_statistics: Dht22Kafka;
	mq135_statistics: Mq135Kafka;
}

export default AppState;
